src/main/java/org/example/learningsystem/btp/destination/converter/DestinationConverter.java

```diff
@@ -0,0 +1,53 @@
+package org.example.learningsystem.btp.destination.converter;
+
+import org.example.learningsystem.btp.destination.annotation.DestinationProperty;
+import org.example.learningsystem.btp.destination.dto.DestinationDto;
+import org.example.learningsystem.btp.destination.exception.MissingNoArgsConstructorException;
+import org.example.learningsystem.btp.destination.exception.MissingPropertyException;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Field;
+
+@Component
+@Profile("cloud")
+public class DestinationConverter {
+
+    public <T> T tryConvert(DestinationDto destination, Class<T> desiredType) {
+        var destinationProperties = destination.getProperties();
+        var requiredFields = desiredType.getDeclaredFields();
+        var converted = createObject(desiredType);
+
+        for (var requiredField : requiredFields) {
+            var destinationPropertyName = requiredField.getAnnotation(DestinationProperty.class);
+            if (destinationPropertyName != null) {
+                var propertyValue = destinationPropertyName.value();
+                if (!destinationProperties.containsKey(propertyValue)) {
+                    throw new MissingPropertyException(desiredType.getSimpleName(), propertyValue);
+                }
+
+                setValueOfField(requiredField, converted, destinationProperties.get(propertyValue));
+            }
+        }
+        return converted;
```

Comment:
idea is cool, reflection api is used, but it's too complex solution for simple mapping :(
you can keep it if you wish, but reflection ideally should not be used in business logic (I mean when you're implementing it by yourself), in most cases it's used only in libraries/starters